clouddriver-integration/src/test/java/com/netflix/spinnaker/clouddriver/PostgresMigrationContainerTest.java

```diff
@@ -0,0 +1,145 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver;
+
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.time.Duration;
+import java.util.Map;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@Testcontainers
+public class PostgresMigrationContainerTest extends BaseContainerTest {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(PostgresMigrationContainerTest.class);
+
+  private static final String POSTGRES_NETWORK_ALIAS = "postgresHost";
+
+  private static final int POSTGRES_PORT = 5432;
+
+  private PostgreSQLContainer<?> postgres;
+
+  private GenericContainer<?> clouddriverInitialContainer;
+
+  private static DockerImageName previousDockerImageName;
+
+  private String jdbcUrl = "";
+
+  @BeforeAll
+  static void setupInitial() {
+    // using the latest image that is still running on liquibase 3.10.3 to create the conditions
+    // similar to real scenario
+    String fullDockerImageName =
+        "us-docker.pkg.dev/spinnaker-community/docker/clouddriver:5.82.2-dev-release-1.32.x-7a8e6e8b3-202406051721-unvalidated";
+    // Skip the tests if there's no docker image.  This allows gradlew build to work.
+    assumeTrue(fullDockerImageName != null);
+    previousDockerImageName = DockerImageName.parse(fullDockerImageName);
+  }
+
+  @BeforeEach
+  void setup() throws Exception {
+    postgres =
+        new PostgreSQLContainer<>("postgres:15")
+            .withDatabaseName("clouddriver")
+            .withUsername("postgres")
+            .withPassword("postgres")
+            .withNetwork(network)
+            .withNetworkAliases(POSTGRES_NETWORK_ALIAS)
+            .withInitScript("postgres_init.sql")
+            .withReuse(true);
+    postgres.start();
+    jdbcUrl =
+        String.format("jdbc:postgresql://%s:%d/clouddriver", POSTGRES_NETWORK_ALIAS, POSTGRES_PORT);
+
+    // Start the first clouddriver(from previous release) container so that all the db changelog
+    // sets are executed
+    clouddriverInitialContainer =
+        new GenericContainer(previousDockerImageName)
+            .withNetwork(network)
+            .withExposedPorts(7002)
```

Comment:
change CLOUDDRIVER_PORT to protected in BaseContainerTest and use it here?

Reply:
will do